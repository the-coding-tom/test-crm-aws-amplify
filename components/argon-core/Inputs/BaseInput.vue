<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
