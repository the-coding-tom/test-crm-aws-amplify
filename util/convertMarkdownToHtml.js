export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
