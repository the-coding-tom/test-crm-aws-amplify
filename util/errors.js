export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
