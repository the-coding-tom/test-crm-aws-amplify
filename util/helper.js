const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
