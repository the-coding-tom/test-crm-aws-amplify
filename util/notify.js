const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
