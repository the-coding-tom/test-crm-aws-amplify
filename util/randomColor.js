const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
