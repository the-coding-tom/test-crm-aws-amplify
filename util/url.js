export const getQueryParams = url => {
  const urlParam = require('url')

  const parsedUrl = urlParam.parse(url)

  return parsedUrl.search
}
