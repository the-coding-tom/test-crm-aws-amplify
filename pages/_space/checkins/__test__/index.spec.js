import IndexPage from '@/pages/_space/checkins/index.vue'
import SettingsPage from '@/pages/_space/checkins/settings.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BaseHeader from '~/components/argon-core/BaseHeader'
import BootstrapVue from 'bootstrap-vue'
import BasePagination from '~/components/argon-core/BasePagination'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'

import allMemberships from '@/pages/_space/memberships/__test__/resources/all-memberships.json'

describe('Page: /:space/checkins/ | Checkins Index Page', () => {
  let NuxtStore, store, wrapper, $membership, $checkin, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $membership = {
      getAllMemberships: jest.fn().mockResolvedValue({
        data: cloneDeep(allMemberships),
        meta: {},
        links: {}
      })
    }

    $checkin = {
      checkins: jest.fn().mockResolvedValue({ data: [], meta: {}, links: {} }),
      checkin: jest.fn().mockResolvedValue({ data: {} })
    }
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()

    router = new VueRouter({
      routes: [
        {
          name: 'space-checkins-settings',
          component: SettingsPage,
          path: '_space/checkins/settings'
        }
      ]
    })

    const asyncDataResponse = await IndexPage.asyncData({
      store,
      $checkin,
      $membership
    })

    wrapper = createWrapper(
      IndexPage,
      store,
      router,
      {
        searching: false,
        type: 0,
        first_name: '',
        last_name: '',
        email: '',
        company: '',
        email: '',
        membership_id: '',
        loading: false,
        meeting_guest: false,
        ...asyncDataResponse
      },
      {
        stubs: {
          transition: false
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(BasePagination.name, BasePagination)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.component(BaseHeader.name, BaseHeader)
        vueInstance.prototype.$checkin = $checkin
        vueInstance.prototype.$membership = $membership
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('checkin member', async () => {
    wrapper.find('button.btn.btn-primary').trigger('click')
    await wrapper.vm.$nextTick()
    await flushPromises()

    //  select member
    const memberOptions = wrapper.vm.$el.querySelectorAll(
      '.member-checkin-form .el-select-dropdown__item'
    )

    const evt = document.createEvent('MouseEvents')
    evt.initEvent('mouseenter')
    memberOptions[1].dispatchEvent(evt)
    memberOptions[1].click()
    await flushPromises()

    // mock location reload
    const { location } = window
    delete window.location
    window.location = { ...location, reload: jest.fn() }

    // submit form
    wrapper.find('form.member-checkin-form').trigger('submit.prevent')

    expect($checkin.checkin).toHaveBeenCalled()
    expect($checkin.checkin.mock.calls[0][0].type).toBe('member')
    expect($checkin.checkin.mock.calls[0][0].membership_id).toBe(
      allMemberships[1].id
    )
  })

  it('checkin guest', async () => {
    wrapper.find('button.btn.btn-primary').trigger('click')
    await wrapper.vm.$nextTick()
    await flushPromises()

    //  select member
    const memberOptions = wrapper.vm.$el.querySelectorAll(
      '.guest-checkin-form .el-select-dropdown__item'
    )

    const evt = document.createEvent('MouseEvents')
    evt.initEvent('mouseenter')
    memberOptions[0].dispatchEvent(evt)
    memberOptions[0].click()
    await flushPromises()

    const guestDetails = {
      firstName: 'Kim',
      lastName: 'Conveni',
      email: 'kimconveni@ui.com',
      company: 'UI LLC'
    }

    wrapper
      .find('.guest-checkin-form #firstName')
      .setValue(guestDetails.firstName)
    wrapper
      .find('.guest-checkin-form #lastName')
      .setValue(guestDetails.lastName)
    wrapper.find('.guest-checkin-form #email').setValue(guestDetails.email)
    wrapper.find('.guest-checkin-form #company').setValue(guestDetails.company)
    wrapper.find('.guest-checkin-form #meetingGuest').setChecked(true)
    await wrapper.vm.$nextTick()

    // mock location reload
    const { location } = window
    delete window.location
    window.location = { ...location, reload: jest.fn() }

    // submit form
    wrapper.find('form.guest-checkin-form').trigger('submit.prevent')
    await wrapper.vm.$nextTick()

    expect($checkin.checkin).toHaveBeenCalled()
    expect($checkin.checkin.mock.calls[1][0].type).toBe('guest')
    expect($checkin.checkin.mock.calls[1][0].membership_id).toBe(
      allMemberships[0].id
    )
    expect($checkin.checkin.mock.calls[1][0].first_name).toBe(
      guestDetails.firstName
    )
    expect($checkin.checkin.mock.calls[1][0].last_name).toBe(
      guestDetails.lastName
    )
    expect($checkin.checkin.mock.calls[1][0].email).toBe(guestDetails.email)
    expect($checkin.checkin.mock.calls[1][0].company).toBe(guestDetails.company)
    expect($checkin.checkin.mock.calls[1][0].meeting_guest).toBe(true)
  })
})
