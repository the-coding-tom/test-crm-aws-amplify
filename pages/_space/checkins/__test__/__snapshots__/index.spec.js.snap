// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/checkins/ | Checkins Index Page renders properly 1`] = `
<div>
  <div class="header pb-6">
    <div class="container-fluid">
      <div class="header-body">
        <div class="d-flex justify-content-between align-items-center py-4">
          <h1 class="page-title"><span>Members</span> <b>•</b> Check-in</h1>
          <div><a href="#_space/checkins/settings" class="btn btn-transparent" target="_self"><i class="fa fa-cogs"></i> Settings</a> <button type="button" class="btn btn-primary">Manual Check-in</button></div>
        </div>
      </div>
    </div>
  </div>
  <div class="container-fluid mt--6">
    <div class="row">
      <div class="col">
        <div class="card">
          <div class="el-table table table-hover table-striped table-responsive el-table--fit el-table--scrollable-x el-table--enable-row-hover">
            <div class="hidden-columns">
              <div></div>
              <div></div>
              <div></div>
              <div></div>
              <div></div>
              <div></div>
            </div>
            <div class="el-table__header-wrapper">
              <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 500px;">
                <colgroup>
                  <col name="el-table_1_column_1" width="80">
                  <col name="el-table_1_column_2" width="80">
                  <col name="el-table_1_column_3" width="80">
                  <col name="el-table_1_column_4" width="80">
                  <col name="el-table_1_column_5" width="100">
                  <col name="el-table_1_column_6" width="80">
                </colgroup>
                <thead class="">
                  <tr class="thead-light">
                    <th colspan="1" rowspan="1" class="el-table_1_column_1     is-leaf is-sortable">
                      <div class="cell">Name / Company<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_2     is-leaf is-sortable">
                      <div class="cell">Type<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_3     is-leaf is-sortable">
                      <div class="cell">For<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_4     is-leaf is-sortable">
                      <div class="cell">Meeting Guest<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_5     is-leaf is-sortable">
                      <div class="cell">Timestamp: Check-in / Check-out<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_6     is-leaf is-sortable">
                      <div class="cell">Check<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                  </tr>
                </thead>
              </table>
            </div>
            <div class="el-table__body-wrapper is-scrolling-left">
              <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 500px;">
                <colgroup>
                  <col name="el-table_1_column_1" width="80">
                  <col name="el-table_1_column_2" width="80">
                  <col name="el-table_1_column_3" width="80">
                  <col name="el-table_1_column_4" width="80">
                  <col name="el-table_1_column_5" width="100">
                  <col name="el-table_1_column_6" width="80">
                </colgroup>
                <tbody>
                  <!---->
                </tbody>
              </table>
              <div class="el-table__empty-block" style="height: 100%; width: 500px;"><span class="el-table__empty-text">暂无数据</span></div>
              <!---->
            </div>
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="el-table__column-resize-proxy" style="display: none;"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div id="checkin___BV_modal_outer_" style="position: absolute; z-index: 1040;">
    <div id="checkin" role="dialog" aria-hidden="true" aria-labelledby="checkin___BV_modal_title_" aria-describedby="checkin___BV_modal_body_" class="modal fade" style="display: none;">
      <div class="modal-dialog modal-md">
        <!---->
        <div id="checkin___BV_modal_content_" tabindex="-1" class="modal-content">
          <header id="checkin___BV_modal_header_" class="modal-header">
            <h5 id="checkin___BV_modal_title_" class="modal-title">Checkin</h5><button type="button" aria-label="Close" class="close">×</button>
          </header>
          <div id="checkin___BV_modal_body_" class="modal-body">
            <div class="tabs nav-fill flex-column flex-sm-row" id="__BVID__25">
              <!---->
              <div class="card-header">
                <ul role="tablist" class="nav nav-pills card-header-pills" id="__BVID__25__BV_tab_controls_">
                  <!---->
                  <li role="presentation" class="nav-item"><a role="tab" aria-selected="true" aria-setsize="2" aria-posinset="1" href="#" target="_self" class="nav-link active" id="__BVID__26___BV_tab_button__" aria-controls="__BVID__26">Member</a></li>
                  <li role="presentation" class="nav-item"><a role="tab" tabindex="-1" aria-selected="false" aria-setsize="2" aria-posinset="2" href="#" target="_self" class="nav-link" id="__BVID__38___BV_tab_button__" aria-controls="__BVID__38">Guest</a></li>
                  <!---->
                </ul>
              </div>
              <div class="tab-content" id="__BVID__25__BV_tab_container_">
                <div role="tabpanel" aria-hidden="false" class="tab-pane active card-body" id="__BVID__26" aria-labelledby="__BVID__26___BV_tab_button__">
                  <form class="member-checkin-form">
                    <fieldset class="form-group" id="__BVID__28">
                      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__28__BV_label_">Member</legend>
                      <div tabindex="-1" role="group" class="bv-no-focus-ring">
                        <div class="el-select" required="required">
                          <!---->
                          <div class="el-input el-input--suffix">
                            <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="Choose a member" class="el-input__inner">
                            <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-"></i><!----><!----><!----><!----><!----></span>
                            <!----></span>
                            <!---->
                            <!---->
                          </div>
                          <div class="el-select-dropdown el-popper" style="display: none;">
                            <div class="el-scrollbar" style="">
                              <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                                <ul class="el-scrollbar__view el-select-dropdown__list">
                                  <!---->
                                  <li class="el-select-dropdown__item"><span>Ashwin Ravichandran</span></li>
                                  <li class="el-select-dropdown__item"><span>Bubu Nyavor</span></li>
                                </ul>
                              </div>
                              <div class="el-scrollbar__bar is-horizontal">
                                <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                              </div>
                              <div class="el-scrollbar__bar is-vertical">
                                <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                              </div>
                            </div>
                            <!---->
                          </div>
                        </div>
                        <!---->
                        <!---->
                        <!---->
                      </div>
                    </fieldset> <button type="submit" class="btn float-right member-checkin-btn btn-primary">Checkin</button>
                  </form>
                </div>
                <div role="tabpanel" aria-hidden="true" class="tab-pane card-body" style="display: none;" id="__BVID__38" aria-labelledby="__BVID__38___BV_tab_button__">
                  <form class="guest-checkin-form">
                    <fieldset class="form-group" id="__BVID__40">
                      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__40__BV_label_">Member</legend>
                      <div tabindex="-1" role="group" class="bv-no-focus-ring">
                        <div class="el-select" required="required">
                          <!---->
                          <div class="el-input el-input--suffix">
                            <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="Choose a member" class="el-input__inner">
                            <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-"></i><!----><!----><!----><!----><!----></span>
                            <!----></span>
                            <!---->
                            <!---->
                          </div>
                          <div class="el-select-dropdown el-popper" style="display: none;">
                            <div class="el-scrollbar" style="">
                              <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                                <ul class="el-scrollbar__view el-select-dropdown__list">
                                  <!---->
                                  <li class="el-select-dropdown__item"><span>Ashwin Ravichandran</span></li>
                                  <li class="el-select-dropdown__item"><span>Bubu Nyavor</span></li>
                                </ul>
                              </div>
                              <div class="el-scrollbar__bar is-horizontal">
                                <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                              </div>
                              <div class="el-scrollbar__bar is-vertical">
                                <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                              </div>
                            </div>
                            <!---->
                          </div>
                        </div>
                        <!---->
                        <!---->
                        <!---->
                      </div>
                    </fieldset>
                    <fieldset class="form-group" id="__BVID__50">
                      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__50__BV_label_">First Name</legend>
                      <div tabindex="-1" role="group" class="bv-no-focus-ring"><input id="firstName" type="text" placeholder="John" required="required" aria-required="true" class="form-control">
                        <!---->
                        <!---->
                        <!---->
                      </div>
                    </fieldset>
                    <fieldset class="form-group" id="__BVID__52">
                      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__52__BV_label_">Last Name</legend>
                      <div tabindex="-1" role="group" class="bv-no-focus-ring"><input id="lastName" type="text" placeholder="Doe" required="required" aria-required="true" class="form-control">
                        <!---->
                        <!---->
                        <!---->
                      </div>
                    </fieldset>
                    <fieldset class="form-group" id="__BVID__54">
                      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__54__BV_label_">Email</legend>
                      <div tabindex="-1" role="group" class="bv-no-focus-ring"><input id="email" type="email" placeholder="john@doe.com" required="required" aria-required="true" class="form-control">
                        <!---->
                        <!---->
                        <!---->
                      </div>
                    </fieldset>
                    <fieldset class="form-group" id="__BVID__56">
                      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__56__BV_label_">Company</legend>
                      <div tabindex="-1" role="group" class="bv-no-focus-ring"><input id="company" type="text" placeholder="SHACK15" required="required" aria-required="true" class="form-control">
                        <!---->
                        <!---->
                        <!---->
                      </div>
                    </fieldset>
                    <div class="custom-control custom-checkbox"><input id="meetingGuest" type="checkbox" autocomplete="off" class="custom-control-input" value="true"><label for="meetingGuest" class="custom-control-label"> Is a meeting guest</label></div> <button type="submit" class="btn float-right btn-primary">Checkin</button>
                  </form>
                </div>
                <!---->
              </div>
            </div>
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
    <!---->
  </div>
  <div>
    <ul class="pagination justify-content-center">
      <li class="page-item prev-page disabled"><a aria-label="Previous" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-left"></i></span></a></li>
      <li class="page-item active"><a class="page-link">1</a></li>
      <li class="page-item next-page disabled"><a aria-label="Next" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-right"></i></span></a></li>
    </ul>
  </div>
</div>
`;
